File: src/domain/valueObjects/post-title.ts
export class PostTitle {
    private readonly _value: string;

    private constructor(value: string) {
        this._value = value;
    }

    get value(): string {
        return this._value;
    }

    static create(value: string): PostTitle {
        if (!value || value.trim().length === 0) {
            throw new Error('PostTitle cannot be empty');
        }
        if (value.length > 255) {
            throw new Error('PostTitle cannot exceed 255 characters');
        }
        return new PostTitle(value);
    }
}

File: lib/blog/domain/value_objects/post_title.dart
class PostTitle {
  final String value;

  PostTitle._({required this.value});

  factory PostTitle.create(String raw) {
    if (raw.trim().isEmpty) throw ArgumentError('PostTitle cannot be empty');
    return PostTitle._(value: raw.trim());
  }

  factory PostTitle.empty() => PostTitle._(value: '');

  @override
  bool operator ==(Object other) => other is PostTitle && other.value == value;

  @override
  int get hashCode => value.hashCode;
}
